internal/push/rest/rest_pusher.go

```diff
@@ -84,53 +89,49 @@ func NewRestPusher(config RestPusherConfig, checkID string, logger *log.Entry) *
 	if config.BufferLen == 0 {
 		config.BufferLen = defaultPushMsgBufferLen
 	}
-	r := &RestPusher{
+	r := &Pusher{
 		c:          client,
 		checkID:    checkID,
 		msgsToSend: make(chan pusherMsg, config.BufferLen),
 		logger:     logger,
-		finished:   &sync.WaitGroup{},
+		finished:   make(chan error, 1),
 	}
 	// The wg only has to monitor pusher state
-	r.finished.Add(1)
 	goPusher(r.msgsToSend, client, logger.WithField("subcomponent", "gopusher"), r.finished)
 	logger.Debug("Creating NewRestPusher created")
 	return r
 }
 
-/* Pusher loops over buffered channel. Range only exits when the channel
-is closed. */
-func goPusher(c chan pusherMsg, client *resty.Client, l *log.Entry, wg *sync.WaitGroup) {
+// Pusher loops over buffered channel. Range only exits when the channel is
+// closed.
+func goPusher(c chan pusherMsg, client *resty.Client, l *log.Entry, done chan<- error) {
 	go func() {
-		// NOTE: race condition found #2
-		// NOTE: race condition found #3
+		var err error
 		l.Debug("goPusher running")
-		defer wg.Done()
+		defer func() { done <- err }()
```

Comment:
Is it OK that we send just the `err` for the last iteration of the range, right?

Reply:
Yes I think it is enough to just capture one of the possible errors, concretely the last one as you noted.